src/components/inputs/registration-company-input.js

```diff
@@ -0,0 +1,140 @@
+/**
+ * Copyright 2017 OpenStack Foundation
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+import React from 'react';
+import AsyncSelect from 'react-select/lib/Async';
+import { queryRegistrationCompanies } from '../../utils/query-actions';
+
+export default class RegistrationCompanyInput extends React.Component {
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            freeInput: false,
+            inputValue: '',
+            noOptions: false,
+        }
+
+        this.handleChange = this.handleChange.bind(this);
+        this.getCompanies = this.getCompanies.bind(this);
+        this.handleInputChange = this.handleInputChange.bind(this);
+
+    }
+
+    handleChange(value) {
+        const isMulti = (this.props.hasOwnProperty('multi') || this.props.hasOwnProperty('isMulti'));
+        const theValue = isMulti ? value.map(v => ({ id: v.value, name: v.label })) : { id: value.value, name: value.label };
+
+        let ev = {
+            target: {
+                id: this.props.id,
+                value: theValue,
+                type: 'companyinput'
+            }
+        };
+
+        if (value.value === null) {
+            this.setState({ freeInput: true })
+        }
+
+        this.props.onChange(ev);
+    }
+
+    handleInputChange(ev) {
+        this.setState({ inputValue: ev.target.value }, () => {
+            let ev = {
+                target: {
+                    id: this.props.id,
+                    value: { id: null, name: this.state.inputValue },
+                    type: 'companyinput'
+                }
+            };
+            this.props.onChange(ev);
+        });
+    }
+
+    getCompanies(input, callback) {
+        let { summitId } = this.props;
+
+        // we need to map into value/label because of a bug in react-select 2
+        // https://github.com/JedWatson/react-select/issues/2998
+
+        const translateOptions = (options) => {            
+            if (options instanceof Error) {
+                this.props.onError(options);
```

Comment:
please lets do this.props.onError optional or lets define the prop types and put it as required